tests/http/server.cpp

```diff
@@ -269,6 +272,17 @@ class Client : public HTTPClient
         const auto& responseHeaders = headers(response);
         const size_t responseHeaderSize =
             std::distance(responseHeaders.begin(), responseHeaders.end());
+
+        // Print headers before error to help debugging
+        if (responseHeaderSize != expectedHeaders.size())
+        {
+            std::cout << "--- Response headers ---" << std::endl;
+            for (const auto& header : responseHeaders)
+                std::cout << header.first << ": " << header.second << std::endl;
+            std::cout << "--- Expected headers ---" << std::endl;
+            for (const auto& header : expectedHeaders)
+                std::cout << header.first << ": " << header.second << std::endl;
+        }
```

Comment:
Is this something that you want to keep? Or is it a left over from debugging sessions?